src/main/kotlin/io/snyk/plugin/services/download/CliDownloaderErrorHandler.kt

```diff
@@ -14,16 +16,30 @@ class CliDownloaderErrorHandler {
         SnykBalloonNotificationHelper.showError(message, project,
             NotificationAction.createSimple("Retry CLI download") {
                 project.getService(SnykCliDownloaderService::class.java).downloadLatestRelease(indicator, project)
+                runBackgroundableTask("Retry Snyk CLI Download", project, false) {
+                    project.getService(SnykCliDownloaderService::class.java).downloadLatestRelease(it, project)
+                }
             },
             NotificationAction.createSimple("Contact support...") {
                 BrowserUtil.browse("https://snyk.io/contact-us/?utm_source=JETBRAINS_IDE")
             })
     }
 
     fun handleIOException(exception: IOException, indicator: ProgressIndicator, project: Project) {
-        val downloader = project.getService(SnykCliDownloaderService::class.java).downloader
-        downloader.downloadFile(getCliFile(), indicator)
-        showErrorWithRetryAndContactAction(getNetworkErrorNotificationMessage(exception), indicator, project)
+        retryDownload(project, indicator, getNetworkErrorNotificationMessage(exception))
+    }
+
+    private fun retryDownload(project: Project, indicator: ProgressIndicator, message: String) {
+        try {
+            runBackgroundableTask("Retry CLI Download", project, false) {
+                val downloader = project.getService(SnykCliDownloaderService::class.java).downloader
+                downloader.downloadFile(getCliFile(), indicator)
+            }
```

Comment:
duplicate of the runBackgroundableTask from above. Shell  we extract fun here?

Reply:
It's not duplicate - slight difference, added comment.